inloop/solutions/templates/solutions/solution_detail.html

```diff
@@ -50,6 +50,12 @@
     </p>
   {# NOTE: states "lost" and "error" are handled in the view code #}
   {% endif %}
+  {% for upfile in upfiles %}
+    <h3 class="inline">{{ upfile.title }}</h3><button type="button" class="solution-show" data-toggle="collapse" data-target="#collapse{{ forloop.counter }}">
+      Show file
+    </button><br>
+    <pre id="collapse{{ forloop.counter }}" class="collapse"><code>{{ upfile.code }}</code></pre><br>
+  {% endfor %}
```

Comment:
Don't forget the maximum line length of 100 chars. It is easier to view diffs if the lines are correctly wrapped.